lib/board.rb

```diff
@@ -0,0 +1,54 @@
+class Board
+
+  attr_accessor :cells
+
+  @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+
+  def initialize
+    reset!
+  end
+
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    row0 = " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    row1 = " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    row2 = " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    lines = '-----------'
+    spaces = '           '
+    puts row0, lines, row1, lines, row2, spaces
+  end
+
+  def position(user_input)
+    num = user_input.to_i - 1
+      self.cells[num]
+  end
+
+  def full?
+    switch = true
+    self.cells.each {|item| item != " " ? switch : switch = false}
```

Comment:
Use the most expressive and efficient iterator for a problem. `Board#full?` is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

`cells.all? {|cell| cell != " "}`

This is also more efficient in that the loop will stop after the iteration for which a cell is not empty.